### ISSUE TITLE
Evolve Tetris Players Using TUG

### ISSUE THREAD

**Issue by schrum2** [schrum2/MM-NEAT#135]
Targeting Unachieved Goals (TUG) is an algorithm I created that is used in this paper, https://suscholar.southwestern.edu/handle/11214/150 , and in my dissertation, http://nn.cs.utexas.edu/?schrum:phd2014 (page 43 labelled as page 27), but no where else. Therefore, I would love to see it succeed somewhere else, and Tetris might be worth trying.

Specifically, start by evolving with 4 objectives: num single lines, num double lines, num triple lines, num Tetrises. This is an interesting baseline to try in its own right. You will need to add support for this in:
https://github.com/schrum2/MM-NEAT/blob/master/src/main/java/edu/southwestern/tasks/rlglue/tetris/TetrisTask.java

However, the next step is to do the same thing, but using TUG. The TUG algorithms exists in my code, and some examples of how to use it in Ms. Pac-Man are in  https://github.com/schrum2/MM-NEAT/tree/master/batch/Experiments-2014-Dissertation-MsPacMan

Note: Once the code part is finished, you will need to make batch files to evolve in Tetris. Make some using plain NEAT with intelligent features, and CNN-HyperNEAT with raw features (depth 1 width 1, MSS, threshold link encoding), since these are the two approaches that we know work best.

**Comment by DevonFulcher** [comment]
Couldn't find the parameter for MSS but it is on by default, yes?

**Comment by schrum2** [comment]
Yes, MSS is the default. To use global coordinates, you set substrateLocationInputs:true and substrateBiasLocationInputs:true

**Comment by DevonFulcher** [comment]
TUG seems to be working. I post watched and it seemed to be fine but maybe we should post watch together to see if you catch anything that is out of the ordinary.

**Comment by DevonFulcher** [comment]
Leaving open until we fully understand TUG behavior.

**Comment by schrum2** [comment]
Devon basically finished and tested this, but I thought it looked weird and wanted to analyze it further. However, actually getting around to doing that is fairly low priority.
